### PR TITLE
Fix multiphase datastore problem

### PR DESCRIPTION
**Pull request by chicm-ms**

### CODE REVIEW

File: src/nni_manager/core/nniDataStore.ts [hunk 1]
```diff
@@ -161,9 +165,8 @@ class NNIDataStore implements DataStore {
 
     private async getFinalMetricData(trialJobId: string): Promise<any> {
         const metrics: MetricDataRecord[] = await this.getMetricData(trialJobId, 'FINAL');
-        assert(metrics.length <= 1);
-        if (metrics.length === 1) {
-            return metrics[0];
+        if (metrics.length > 0) {
+            return metrics[metrics.length - 1];
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: Should we also consider to log error if metrics.length > 1 for single phase exp?

Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Ok, added back the assertion for single phase exp.

File: src/nni_manager/core/nniDataStore.ts [hunk 2]
```diff
@@ -161,11 +167,31 @@ class NNIDataStore implements DataStore {
...
             return metrics[0];
+        }
+    }
+
+    private async isMultiPhase(): Promise<boolean|undefined> {
+        if (this.multiPhase === undefined) {
+            this.multiPhase = (await this.getExperimentProfile(getExperimentId())).params.multiPhase;
+        }
+
+        if (this.multiPhase !== undefined) {
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: Optional: change to 
if (this.multiPhase) { 
}
for simplicity

File: src/nni_manager/core/nniDataStore.ts [hunk 3]
```diff
@@ -161,11 +167,31 @@ class NNIDataStore implements DataStore {
...
+        const multiPhase: boolean | undefined = await this.isMultiPhase();
+
+        if (multiPhase) {
+            if (metrics.length > 0) {
+                return metrics[metrics.length - 1];
+            } else {
+                return undefined;
+            }
+        } else {
+            assert(metrics.length <= 1);
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: Log error here, rather than assert

File: src/nni_manager/core/nniDataStore.ts (hunk 2 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Fixed.

File: src/nni_manager/core/nniDataStore.ts (hunk 3 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Fixed.

**Pull request merged**